lighthouse-core/gather/gatherers/dobetterweb/optimized-images.js

```diff
@@ -101,16 +103,31 @@ class OptimizedImages extends Gatherer {
    * @return {!Promise<{originalSize: number, jpegSize: number, webpSize: number}>}
    */
   calculateImageStats(driver, networkRecord) {
-    const param = JSON.stringify(networkRecord.url);
-    const script = `(${getOptimizedNumBytes.toString()})(${param})`;
-    return driver.evaluateAsync(script).then(stats => {
-      const isBase64DataUri = networkRecord.isBase64DataUri;
-      const base64Length = networkRecord.url.length - networkRecord.url.indexOf(',') - 1;
-      return {
-        originalSize: isBase64DataUri ? base64Length : networkRecord.resourceSize,
-        jpegSize: isBase64DataUri ? stats.jpeg.base64 : stats.jpeg.binary,
-        webpSize: isBase64DataUri ? stats.webp.base64 : stats.webp.binary,
-      };
+    let uriPromise = Promise.resolve(networkRecord.url);
+
+    // Get the request body for cross-origin images to circumvent canvas CORS protections
+    if (!networkRecord.isSameOrigin && !networkRecord.isBase64DataUri) {
+      const requestId = networkRecord.requestId;
+      uriPromise = Promise.resolve()
+        .then(_ => driver.sendCommand('Network.enable'))
```

Comment:
conceptually yes, but we enable network in a couple places without disabling, so I'm somewhat ambivalent. (in fact if we disable network properly elsewhere this might not work... :/)